tasks/main.yml

```diff
@@ -25,12 +25,53 @@
     - install
     - dependencies
 
-- name: Unzip ghosts-agent
+- name: Create ghosts directory and download ghosts v8.0.0
   become: true
   become_user: "{{ ghostsagent_user }}"
-  ansible.builtin.unarchive:
-    src: ghosts.tar.gz
-    dest: "{{ ghostagent_path }}"
+  ansible.builtin.shell:
+    cmd: |
+      mkdir -p ~/ghosts
+      wget https://cmu.box.com/shared/static/tdozbmmdyvajubohwtnon1huyfqjuwrz.zip -O ~/ghosts/ghosts.zip
+    creates: "~/ghosts/ghosts.zip"
+  tags:
+    - install
+    - ghostagent
+
+- name: Unzip ghosts.zip and ghosts-client-linux-v8.0.0.zip
+  become: true
+  become_user: "{{ ghostsagent_user }}"
+  ansible.builtin.shell:
+    cmd: |
+      unzip ~/ghosts/ghosts.zip -d ~/ghosts && \
+      unzip ~/ghosts/ghosts-client-linux-v8.0.0.zip -d ~/ghosts
+  args:
+    creates: "~/ghosts/ghosts-client-linux-v8.0.0"
+  tags:
+    - install
+    - ghostagent
+
+- name: Clean up zip files
+  become: true
+  become_user: "{{ ghostsagent_user }}"
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: absent
+  with_items:
+    - ~/ghosts/ghosts.zip
+    - ~/ghosts/ghosts-client-linux-v8.0.0.zip
+  tags:
+    - install
+    - ghostagent
+
+- name: Update application.json with the ghostsserver url
+  become: true
+  become_user: "{{ ghostsagent_user }}"
+  ansible.builtin.template:
+    src: application.json.j2
+    dest: "~/ghosts/build_output/config/application.json"
+  tags:
+    - update
+    - ghostagent
```

Comment:
You can simplify that using ansible.builtin.unarchive: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html

Reply:
The downloaded ghostagent zip file has another inner zip file, so I could not simplify it that much, but I'm using now the ansible modules instead of shell commands